### ISSUE TITLE
"classic" theme has incorrect colors

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#4102]
Environment
-----------

*  SuperCollider version: https://github.com/supercollider/supercollider/commit/82e7ba4d1405d3095969652e80ebd00c8719d8ba
*  operating system and version: macOS 10.13.6

Steps to reproduce (for bugs)
-----------------------------

```
cmake -DCMAKE_PREFIX_PATH=`brew --prefix qt5` -DSUPERNOVA=ON ..
```

The QT version is  5.11.2, as installed from homebrew. Cmake finds a softlink to` ../Cellar/qt/5.11.2`.

Current Behavior
----------------

The IDE has mostly black windows.

<img width="510" alt="screen shot 2018-10-01 at 18 46 34" src="https://user-images.githubusercontent.com/1810559/46306454-9c305b00-c5b4-11e8-8989-2dd88d53aa7a.png">

**Comment by snappizz** [comment]
do you have a custom theme installed? if so can you share `sc_ide_conf.yaml`?

**Comment by telephon** [comment]
Not sure, I can't read the preferences anymore, they are too dark ;).

No, sure. I had edited the theme.
Here it is:

```yaml

IDE:
  editor:
    blinkDuration: 30
    disableBlinkingCursor: false
    font:
      antialias: 2
      family: Droid Sans Mono
      size: 10
    highlightBracketContents: true
    highlightCurrentLine: true
    inactiveEditorFadeAlpha: 64
    indentWidth: 4
    insertMatchingTokens: false
    lineWrap: true
    postWindowScrollback: 1000
    showAutocompleteHelp: true
    showLinenumber: true
    showWhitespace: false
    spaceIndent: false
    stepForwardEvaluation: true
    theme: My old theme
    themes:
      My old theme:
        built-in: !textFormat
          color: "#3333bf"
          italic: false
        char: !textFormat
          color: "#007300"
          italic: false
        class: !textFormat
          color: "#0000d2"
          italic: false
        comment: !textFormat
          color: "#bf0000"
          italic: false
        currentLine: !textFormat
          color: "#000000"
          background: "#f4f4f4"
          italic: false
        env-var: !textFormat
          color: "#8c4614"
          italic: false
        evaluatedCode: !textFormat
          color: "#000000"
          background: "#f8a200"
        keyword: !textFormat
          color: "#0000e6"
          bold: true
          italic: false
        lineNumbers: !textFormat
          color: "#848484"
          background: "#ffffff"
          italic: false
        matchingBrackets: !textFormat
          color: "#ff0000"
          background: "#ffe2f9"
          bold: true
          italic: false
        mismatchedBrackets: !textFormat
          color: "#000000"
          background: "#ccb8e2"
          italic: false
        number: !textFormat
          color: "#980099"
          italic: false
        postwindowemphasis: !textFormat
          color: "#000000"
          bold: true
          italic: false
        postwindowerror: !textFormat
          color: "#d11c24"
          italic: false
        postwindowsuccess: !textFormat
          color: "#738a05"
          italic: false
        postwindowtext: !textFormat
          color: "#000000"
          italic: false
        postwindowwarning: !textFormat
          color: "#a57706"
          italic: false
        primitive: !textFormat
          color: "#3333bf"
          italic: false
        searchResult: !textFormat
          color: "#000000"
          background: "#b6d341"
          italic: false
        selection: !textFormat
          {}
        string: !textFormat
          color: "#5f5f5f"
          italic: false
        symbol: !textFormat
          color: "#007300"
          italic: false
        text: !textFormat
          {}
        whitespace: !textFormat
          color: "#7f7f7f"
          italic: false
      droid:
        built-in: !textFormat
          color: "#3333bf"
          italic: false
        char: !textFormat
          color: "#007300"
          italic: false
        class: !textFormat
          color: "#0000d2"
          italic: false
        comment: !textFormat
          color: "#bf0000"
          italic: false
        currentLine: !textFormat
          color: "#000000"
          background: "#f4f4f4"
          italic: false
        env-var: !textFormat
          color: "#ff6600"
          italic: false
        evaluatedCode: !textFormat
          color: "#000000"
          background: "#f8a200"
        keyword: !textFormat
          color: "#0000e6"
          bold: true
          italic: false
        lineNumbers: !textFormat
          color: "#000000"
          italic: false
        matchingBrackets: !textFormat
          {}
        mismatchedBrackets: !textFormat
          color: "#ffffff"
          background: "#fca7c0"
          italic: false
        number: !textFormat
          color: "#980099"
          italic: false
        postwindowemphasis: !textFormat
          color: "#000000"
          bold: true
          italic: false
        postwindowerror: !textFormat
          color: "#d11c24"
          italic: false
        postwindowsuccess: !textFormat
          color: "#738a05"
          italic: false
        postwindowtext: !textFormat
          color: "#000000"
          italic: false
        postwindowwarning: !textFormat
          color: "#a57706"
          italic: false
        primitive: !textFormat
          color: "#3333bf"
          italic: false
        searchResult: !textFormat
          color: "#00ff0b"
          background: "#729fcc"
          italic: false
        selection: !textFormat
          color: "#000000"
          background: "#81ff76"
          italic: false
        string: !textFormat
          color: "#5f5f5f"
          italic: false
        symbol: !textFormat
          color: "#007300"
          italic: false
        text: !textFormat
          {}
        whitespace: !textFormat
          color: "#7f7f7f"
          italic: false
      solarized-dark:
        built-in: !textFormat
          color: "#3333bf"
          italic: false
        char: !textFormat
          color: "#007300"
          italic: false
        class: !textFormat
          color: "#0000d2"
          italic: false
        comment: !textFormat
          color: "#bf0000"
          italic: false
        currentLine: !textFormat
          color: "#000000"
          background: "#f4f4f4"
          italic: false
        env-var: !textFormat
          color: "#ff6600"
          italic: false
        evaluatedCode: !textFormat
          color: "#000000"
          background: "#f8a200"
        keyword: !textFormat
          color: "#0000e6"
          bold: true
          italic: false
        lineNumbers: !textFormat
          color: "#000000"
          italic: false
        matchingBrackets: !textFormat
          {}
        mismatchedBrackets: !textFormat
          color: "#ffffff"
          background: "#fca7c0"
          italic: false
        number: !textFormat
          color: "#980099"
          italic: false
        postwindowemphasis: !textFormat
          color: "#000000"
          bold: true
          italic: false
        postwindowerror: !textFormat
          color: "#d11c24"
          italic: false
        postwindowsuccess: !textFormat
          color: "#738a05"
          italic: false
        postwindowtext: !textFormat
          color: "#000000"
          italic: false
        postwindowwarning: !textFormat
          color: "#a57706"
          italic: false
        primitive: !textFormat
          color: "#3333bf"
          italic: false
        searchResult: !textFormat
          color: "#00ff0b"
          background: "#729fcc"
          italic: false
        selection: !textFormat
          color: "#000000"
          background: "#81ff76"
          italic: false
        string: !textFormat
          color: "#5f5f5f"
          italic: false
        symbol: !textFormat
          color: "#007300"
          italic: false
        text: !textFormat
          {}
        whitespace: !textFormat
          color: "#7f7f7f"
          italic: false
    useComboBox: false
    useComboBoxWhenSplitting: true
  interpreter:
    autoStart: true
    command: sclang
    configFile: /Users/wanninger/Library/Application Support/SuperCollider/sclang_conf_current.yaml
    runtimeDir: ""
    standalone: true
  mainWindow:
    detached: !QVariantMap
      documents-dock: ""
      help-dock: ""
      post-dock: ""
    geometry: AdnQywACAAAAAAAOAAAAFwAABQ0AAAMfAAAADgAAAC0AAAUNAAADHwAAAAAAAAAABQA=
    state: AAAA/wAAAAD9AAAAAgAAAAEAAAI7AAAC2PwCAAAABPsAAAASAHAAbwBzAHQALQBkAG8AYwBrAQAAAAAAAAFlAAAATAD////7AAAAHABkAG8AYwB1AG0AZQBuAHQAcwAtAGQAbwBjAGsBAAABZgAAAGIAAABeAP////sAAAAcAGQAbwBjAHUAbQBlAG4AdABzAC0AZABvAGMAawIAAAHtAAAAFgAAAWQAAAMb+wAAABIAaABlAGwAcAAtAGQAbwBjAGsBAAAByQAAAQ8AAABMAP///wAAAAMAAAWbAAABhvwBAAAAAfsAAAASAHAAbwBzAHQALQBkAG8AYwBrAAAAAAAAAANeAAAAAAAAAAAAAALEAAAC2AAAAAEAAAACAAAAAQAAAAL8AAAAAA==
  postWindow:
    lineWrap: true
    scrollback: 10000
  recentDocuments: !QVariantList
  shortcuts:
    editor-autocompletion: Meta+Space
    editor-copy-line-down: Ctrl+Shift+C
    editor-copy-line-up: Meta+Ctrl+Alt+ៀ
    editor-go-to-next-empty-line: Alt+Num+Down
    editor-go-to-next-region: Ctrl+Num+Down
    editor-go-to-prev-empty-line: Alt+Num+Up
    editor-go-to-prev-region: Ctrl+Num+Up
    editor-move-line-down: Ctrl+Alt+ៀ
    editor-move-line-up: Ctrl+Alt+ៀ
    editor-toggle-comment: Ctrl+Shift+7
    interpreter-recompile-lib: Ctrl+Alt+L
    interpreter-restart: Ctrl+Shift+L
    synth-server-mute: F10
    synth-server-volume-down: F11
    synth-server-volume-up: F12
  startWithSession: last
  useGitHubBugReport: false

```

**Comment by telephon** [comment]
And when I move this out and use a fresh one, I get a new one that is bright again. So not a big issue, but maybe something that a few people might trip over.

**Comment by telephon** [comment]
But when switching between themes, some of the old theme remains active until I reopen the IDE.

**Comment by brianlheim** [comment]
Sounds like the IDE UI colors need to be refreshed after updating the theme. There must already be a callback that runs for the theme, so whatever calls that should call the UI coloring update as well.

**Comment by brianlheim** [comment]
Taking a closer look, it's only when switching to the 'classic' theme that the UI colors aren't applied.

**Comment by snappizz** [comment]
oh, i know why. "classic" theme takes colors from the UI palette, which results in nonsensical colors now that we're changing the UI palette.

**Comment by snappizz** [comment]
fixed in #4161.
